app/v1/services/EnrolmentsAuthService.scala

```diff
@@ -38,18 +40,55 @@ class EnrolmentsAuthService @Inject()(val connector: AuthConnector) {
-      case Some(Organisation) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Organisation", enrolments)
-      case Some(Agent) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Agent", enrolments)
-      case _ =>
-        Logger.warn(s"[AuthorisationService] [authoriseAsClient] Authorisation failed due to unsupported affinity group.")
-        Future.successful(Left(LegacyUnauthorisedError))
-    } recoverWith unauthorisedError
+  def authorised(predicate: Predicate, nrsRequired: Boolean = false)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[AuthOutcome] = {
+    if(!nrsRequired){
+      authFunction.authorised(predicate).retrieve(affinityGroup and allEnrolments) {
+        case Some(Individual) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Individual", enrolments)
+        case Some(Organisation) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Organisation", enrolments)
+        case Some(Agent) ~ enrolments => createUserDetailsWithLogging(affinityGroup = "Agent", enrolments)
+        case _ =>
+          Logger.warn(s"[AuthorisationService] [authoriseAsClient] Authorisation failed due to unsupported affinity group.")
+          Future.successful(Left(LegacyUnauthorisedError))
+      }recoverWith unauthorisedError
+    } else {
+      authFunction.authorised(predicate).retrieve(affinityGroup and allEnrolments
+        and internalId and externalId and agentCode and credentials
+        and confidenceLevel and nino and saUtr and name and dateOfBirth
+        and email and agentInformation and groupIdentifier and credentialRole
+        and mdtpInformation and credentialStrength and loginTimes
+      ) {
+        case affGroup ~ enrolments ~ inId ~ exId ~ agCode ~ creds
+          ~ confLevel ~ ni ~ saRef ~ nme ~ dob
+          ~ eml ~ agInfo ~ groupId ~ credRole
+          ~ mdtpInfo ~ credStrength ~ logins
+          if affGroup.contains(AffinityGroup.Organisation) || affGroup.contains(AffinityGroup.Individual) || affGroup.contains(AffinityGroup.Agent) =>
+
+          // setup dummy data for ITMP data
+          val dummyItmpName: ItmpName = ItmpName(None, None, None)
+          val dummyItmpDob: Option[LocalDate] = None
+          val dummyItmpAddress: ItmpAddress = ItmpAddress(None, None, None, None, None, None, None, None)
+
+          val identityData =
+            IdentityData(
+              inId, exId, agCode, creds,
+              confLevel, ni, saRef, nme, dob,
+              eml, agInfo, groupId,
+              credRole, mdtpInfo, dummyItmpName, dummyItmpDob,
+              dummyItmpAddress, affGroup, credStrength, logins)
+
+          createUserDetailsWithLogging(affinityGroup = affGroup.get.toString, enrolments, Some(identityData))
+        case _ =>
+          Logger.warn(s"[AuthorisationService] [authoriseAsClient] Authorisation failed due to unsupported affinity group.")
```

Comment:
Could we update the [Class][Method] parts of the logging here to match the new naming? I had originally replicated the naming of the old auth service but that makes it difficult to see where errors are coming from.